domain/camera/src/main/java/com/google/jetpackcamera/domain/camera/CameraXCameraUseCase.kt

```diff
@@ -834,6 +881,34 @@ constructor(
             getResolutionSelector(sessionSettings.aspectRatio)
         )
 
+        if (currentSettings.value?.lowLightBoost == LowLightBoost.ENABLED) {
+            Log.d(TAG, "Adding Control AE_MODE_ON_LOW_LIGHT_BOOST_BRIGHTNESS_PRIORITY")
+            val camera2Interop = Camera2Interop.Extender(previewUseCaseBuilder)
+            if (Build.VERSION.SDK_INT >= 35) {
+                camera2Interop.setCaptureRequestOption(
+                    CaptureRequest.CONTROL_AE_MODE,
+                    CameraMetadata.CONTROL_AE_MODE_ON_LOW_LIGHT_BOOST_BRIGHTNESS_PRIORITY
+                )
+                camera2Interop.setSessionCaptureCallback(object : CaptureCallback() {
+                    override fun onCaptureCompleted(
+                        session: CameraCaptureSession,
+                        request: CaptureRequest,
+                        result: TotalCaptureResult
+                    ) {
+                        super.onCaptureCompleted(session, request, result)
+                        val boostState = result.get(CaptureResult.CONTROL_LOW_LIGHT_BOOST_STATE)
```

Comment:
Not sure if this will change anything for your use case but sharing for context:

`LOW_LIGHT_BOOST_STATE` is only active / inactive if `CaptureResult.CONTROL_AE_MODE` is `AE_ON_LOW_LIGHT_BOOST_BRIGHTNESS_PRIORITY`

So if the current camera feature combination wasn't supported when LLB was requested, the result may switch back to AE_MODE_ON

Or if the AE_MODE was not LLB, then maybe hide the moon icon? 

Unless if the UI keeps the moon icon there but in an INACTIVE state? When pressed it will turn on LLB? But that might not be intuitive because the scene could be well lit and still be inactive so it wouldn't be clear to the user when AE_MODE is LLB or auto.